persist/sqlite/sql.go

```diff
@@ -188,3 +189,40 @@ func setDBVersion(tx *txn, version int64) error {
 func jitterSleep(t time.Duration) {
 	time.Sleep(t + time.Duration(rand.Int63n(int64(t/2))))
 }
+
+func queryPlaceHolders(n int) string {
+	if n == 0 {
+		return ""
+	} else if n == 1 {
+		return "?"
+	}
+	var b strings.Builder
+	b.Grow(((n - 1) * 2) + 1) // ?,?
+	for i := 0; i < n-1; i++ {
+		b.WriteString("?,")
+	}
+	b.WriteString("?")
+	return b.String()
+}
+
+func queryArgs[T any](args []T) []any {
+	if len(args) == 0 {
+		return nil
+	}
+	out := make([]any, len(args))
+	for i, arg := range args {
+		out[i] = arg
+	}
+	return out
+}
+
+func queryArgsFunc[T any](args []T, fn func(t T) any) []any {
```

Comment:
Looking at the call sites this could be replaced by `encodeSlice` which calls `encode` on `[]T`